client/contexts/auth/reducer.ts

```diff
@@ -0,0 +1,14 @@
+import { produce } from 'immer';
+
+import { AuthReducer } from './types';
+
+export const reducer: AuthReducer = (state, action) => {
+  switch (action.type) {
+    case 'SET_AUTHED':
+      return produce(state, draftState => {
+        draftState.authed = action.authed;
+      });
+    default:
```

Comment:
nit (personal preference): I like placing a new line between cases that `break`/`return` to make reading easier and to make intentions clearer.

eg.
```js
switch (foo) {
  case 0:
    doSomething(0);
  case 1:
  case 2:
    return doSomething(1);

  case 3:
  case 4:
    doSomething(2);
    break;
    
  case 4:
    doSomething(3);
    break;
    
  default:
    doSomething(4);
}
```